ReactiveCocoa/Swift/Signal.swift

```diff
@@ -680,11 +680,29 @@ public func tryMap<T, U, E>(operation: T -> Result<U, E>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` until `interval`. Then if signal isn't completed yet,
+/// errors with `error` on `scheduler`.
+public func timeoutWithError<T, E>(error: E, afterInterval interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
```

Comment:
Why do we allow `timeoutWithError()` to have an `interval` is exactly `0`? From the outside, it would almost seem to _guarantee_ that this signal sends the error. But knowing the implementation details, it's more like: "send error if the given signal doesn't complete synchronously". In either case, both are kind of suspicious. Can we make remove ambiguity by saying the interval has to be strictly `> 0`?